receiver/vcenterreceiver/processors.go

```diff
@@ -15,25 +15,63 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/vcenterreceiver/internal/metadata"
 )
 
+type DatacenterStats struct {
+	ClusterStatusCount map[types.ManagedEntityStatus]int64
+	HostStats          map[string]map[types.ManagedEntityStatus]int64
+	VMStats            map[string]map[types.ManagedEntityStatus]int64
+	DatastoreCount     int64
+	DiskCapacity       int64
+	DiskFree           int64
+	CPULimit           int64
+	MemoryLimit        int64
+}
+
 // processDatacenterData creates all of the vCenter metrics from the stored scraped data under a single Datacenter
 func (v *vcenterMetricScraper) processDatacenterData(dc *mo.Datacenter, errs *scrapererror.ScrapeErrors) {
 	// Init for current collection
 	now := pcommon.NewTimestampFromTime(time.Now())
-
-	v.processDatastores(now, dc)
+	dcStats := &DatacenterStats{
+		ClusterStatusCount: make(map[types.ManagedEntityStatus]int64),
+		HostStats:          make(map[string]map[types.ManagedEntityStatus]int64),
+		VMStats:            make(map[string]map[types.ManagedEntityStatus]int64),
+	}
+	v.processDatastores(now, dc, dcStats)
 	v.processResourcePools(now, dc, errs)
-	vmRefToComputeRef := v.processHosts(now, dc, errs)
-	vmGroupInfoByComputeRef := v.processVMs(now, dc, vmRefToComputeRef, errs)
-	v.processClusters(now, dc, vmGroupInfoByComputeRef, errs)
+	vmRefToComputeRef := v.processHosts(now, dc, dcStats, errs)
+	vmGroupInfoByComputeRef := v.processVMs(now, dc, vmRefToComputeRef, dcStats, errs)
+	v.processClusters(now, dc, vmGroupInfoByComputeRef, dcStats, errs)
+	v.buildDatacenterMetrics(now, dc, dcStats)
+}
+
+// buildDatacenterMetrics builds a resource and metrics for a given scraped vCenter Datacenter
+func (v *vcenterMetricScraper) buildDatacenterMetrics(
+	ts pcommon.Timestamp,
+	dc *mo.Datacenter,
+	dcStats *DatacenterStats,
+) {
+	// Create Datastore resource builder
```

Comment:
nit: this has gotten me a number of times, but this and the next comment shouldu refer to Datacenters not Datastores